src/index.js

```diff
@@ -1,54 +1,66 @@
 /* eslint-disable linebreak-style */
+// eslint-disable-next-line linebreak-style
+/* eslint-disable no-undef */
+/* eslint-disable linebreak-style */
 /* eslint-disable no-unused-vars */
 /* eslint-disable linebreak-style */
 /* "import/no-unresolved": "off" */
 
 // import _ from 'lodash';
 import './style.css';
 import html from './template.html';
-import Icon from './images/speaker-background.png';
-
-const arr = [
-  {
-    description: 'wash the dishes',
-    completed: 'true',
-    id: 1,
-  },
-  {
-    description: 'cook food',
-    completed: 'false',
-    id: 2,
-  },
-  {
-    description: 'do laundry',
-    completed: 'false',
-    id: 3,
-  },
-];
-
-// eslint-disable-next-line no-return-assign
-arr.map((card) => document.getElementById('text-content').innerHTML
-  += `<section class="checkbox-label-wrapper">
-        <div class="description">
-          <input type="checkbox" id="checkbox" name="checkbox" value="checkbox">
-          <label for="checkbox" id="checkbox-1" class="label">${card.description}</label>
-        </div>
-        <span class="dots">&#8942</span>
-       </section>`);
-
-// function component() {
-//   const element = document.createElement('div');
-
-//   // Lodash, now imported by this script
-//   element.innerHTML = _.join(['Hello', 'webpack'], ' ');
-//   element.classList.add('hello');
-//   // Add the image to our existing div.
-//   const myIcon = new Image();
-//   myIcon.src = Icon;
-
-//   element.appendChild(myIcon);
-
-//   return element;
-// }
-
-// document.body.appendChild(component());
\ No newline at end of file
+import { showTodo } from './modules/UI.js';
+import { updateStatus } from './modules/updateStatus.js';
+import { deleteTask } from './modules/delete.js';
+
+const taskInput = document.querySelector('.placeholder');
+let editId;
+let isEditTask = false;
+
+// geting localstorage todo-list
+window.todos = JSON.parse(localStorage.getItem('todo-list'));
+
+// to show menu
+window.showMenu = (selectedTask) => {
+  // getting task menu div
+  const taskMenu = selectedTask.nextElementSibling;
+  taskMenu.classList.add('show');
+  document.addEventListener('click', (e) => {
+    // need to add condition for e.target.tagName != "I" ;
+    if (e.target !== selectedTask) {
+      taskMenu.classList.remove('show');
+    }
+  });
+};
+
+// to edit task
+window.editTask = (taskId, taskName) => {
+  editId = taskId;
+  isEditTask = true;
+  taskInput.value = taskName;
+};
+
+showTodo(todos);
+window.updateStatus = updateStatus;
+// window.editTask = editTask;
+// window.showMenu = showMenu;
+window.deleteTask = deleteTask;
+
+taskInput.addEventListener('keyup', (e) => {
+  const userTask = taskInput.value.trim();
+  if (e.key === 'Enter' && userTask) {
+    if (!isEditTask) { //  if isEdidTask isn't true
+      todos = !todos ? [] : todos;
+      const taskInfo = { name: userTask, status: 'pending' };
+      todos.push(taskInfo);
```

Comment:
- [ ] Kindly make sure when adding a new task, consider adding an index for the task and the index set to the value of the new array length (i.e. if a 5th task is added to the list, the index of that task should equal 5). Take also into consideration that deleting a task should update all remaining items' indexes, so they represent the current list order and are unique.